extension/src/popup/components/accountHistory/HistoryItem/index.tsx

```diff
@@ -112,169 +109,229 @@ export const HistoryItem = ({
-    isRecipient = to === publicKey && from !== publicKey;
-    paymentDifference = isRecipient ? "+" : "-";
-    PaymentComponent = (
-      <>
-        {paymentDifference}
-        {formatAmount(new BigNumber(amount).toFixed(2, 1))} {destAssetCode}
-      </>
-    );
-    IconComponent = isRecipient ? (
-      <Icon.ArrowDown className="HistoryItem__icon--received" />
-    ) : (
-      <Icon.ArrowUp className="HistoryItem__icon--sent" />
-    );
-    rowText = destAssetCode;
-    dateText = `${isRecipient ? t("Received") : t("Sent")} \u2022 ${date}`;
-    transactionDetailProps = {
-      ...transactionDetailProps,
-      isRecipient,
-      headerTitle: `${
-        isRecipient ? t("Received") : t("Sent")
-      } ${destAssetCode}`,
-      operationText: `${paymentDifference}${new BigNumber(
-        amount,
-      )} ${destAssetCode}`,
-    };
-  } else if (isCreateExternalAccount) {
-    PaymentComponent = <>-{new BigNumber(startingBalance).toFixed(2, 1)} XLM</>;
-    IconComponent = <Icon.ArrowUp className="HistoryItem__icon--sent" />;
-    rowText = "XLM";
-    dateText = `${t("Sent")} \u2022 ${date}`;
-    transactionDetailProps = {
-      ...transactionDetailProps,
-      headerTitle: t("Create Account"),
-      isPayment: true,
-      operation: {
-        ...operation,
-        asset_type: "native",
-        to: account,
-      },
-      operationText: `-${new BigNumber(startingBalance)} XLM`,
-    };
-  } else if (isSorobanTx) {
-    const attrs = getAttrsFromSorobanHorizonOp(operation, networkDetails);
-    const token = tokenBalances.find(
-      (balance) => attrs && balance.contractId === attrs.contractId,
-    );
+  const [txDetails, setTxDetails] = React.useState(transactionDetailPropsBase);
+  const [dateText, setDateText] = React.useState(date);
+  const [rowText, setRowText] = React.useState("");
+  const [IconComponent, setIconComponent] = React.useState(
+    (
+      <Icon.Shuffle className="HistoryItem__icon--default" />
+    ) as React.ReactElement | null,
+  );
+  const [BodyComponent, setBodyComponent] = React.useState(
+    null as React.ReactElement | null,
+  );
+
+  const renderBodyComponent = () => BodyComponent;
+  const renderIcon = () => IconComponent;
 
-    if (!token || !attrs) {
-      rowText = operationString;
-      transactionDetailProps = {
-        ...transactionDetailProps,
-        headerTitle: t("Transaction"),
-        operationText: operationString,
-      };
-    } else if (attrs.fnName === SorobanTokenInterface.mint) {
-      // handle a mint operation, which is similar to a Sent Payment, but with subtle differences
-      const formattedTokenAmount = formatTokenAmount(
-        new BigNumber(attrs.amount),
-        Number(token.decimals),
-      );
-      PaymentComponent = (
-        <>
-          {formattedTokenAmount} {token.symbol}
-        </>
-      );
-      IconComponent = <Icon.ArrowUp className="HistoryItem__icon--sent" />;
+  React.useEffect(() => {
```

Comment:
I needed to make this async to better handle the mint token case, explained in the PR description.